### ISSUE TITLE
Storing vocabulary terms borrowed from other CVs?

### ISSUE THREAD

**Issue by spficklin** [GMOD/Chado#68]
Ontologies are increasingly borrowing terms from other ontologies and using the IDs from the source ontology, and  the current tables of Chado make this difficult to represent.   One example is the [Plant Trait Ontology](http://www.obofoundry.org/ontology/to.html).  Many of it's terms come from GO, CHEBI and BFO.   I'm working on improvements to the Tripal OBO loader to deal with this and I do not know how we should store these.  

 Consider the case where a GO term is used in the TO.  How do I represent that term?
  
**Option #1**:  Create a new cvterm that duplicates the original GO Term.  Problems:
- If the term changes in the GO then the two records get out of sync.
- This does not accurately reflect the fact that the term belongs to the GO and not the TO. It's just being borrowed.  
- This option forces me to add a dbxref record, and following the foreign keys all the way to the db will lead me to the TO record in the db table.  This has the wrong URL and URLprefix for a GO term.

**Option #2**:  Use the original cvterm from the GO.  This seems like the correct way to go.... but:
-  The relationships for this term may not be the same in borrowing vocabulary.    We cannot represent a new set of relationships that are different from those already created. (e.g. those created by the GO and distinguish those as belonging to the TO).
- If we do not physically add a record to the cvterm table than many existing tools that expect to find all terms for a vocabulary in the cvterm table won't find these borrowed terms.
-  If a borrowed term serves as a root term in the vocabulary how can tools determine that it serves as a root because the cvterm table has a cv_id that points to the owning ontology.  

I don't yet have a solution for this but if I can come up with one I'll post it here.

**Comment by bradfordcondon** [comment]
I'll point out that if we're looking at the GO ontology, we dont care if the TO says a certain relationship exists.  However, if we're looking at the TO, we *do*care.  And we need some way of keeping track of which ontologies say which relationships exist.

For this reason It appears to make sense to have cvterm_relationships also have a CV_id.  In most cases this is the cv of the cvterm's and unnecessary/duplication of information, so it should be nullable?

**Comment by scottcain** [comment]
Option 2 feels really wrong to me, while I realize that option 1 has significant downsides.  While the terms are "borrowed", when they are used in the borrowing ontologies context, we should probably use them as though they belong to the borrowing ontology.  I wonder if @cmungall has any thoughts on this situation, since I imagine he and his group have come up with a solution for themselves.

**Comment by cmungall** [comment]
Yes indeed.

TL;DR use your number2 option above. Note that TO may provide edges coming from TO to GO, but TO should never assert anything *about* a GO ID.

This is a general issue regardless of Chado, and it's not unlike issues faced with dependencies of software libs. Modern ontology development is modular, with TO reusing PO, GO etc and providing links.

Historically we have offered a number of different artefacts in ad-hoc ways. Sometimes the ontology will have portions of other ontologies merged in. Sometimes these modules will be imported using owl:imports. Sometimes the external stuff is just stripped out.

What I think you want to load in Chado is what we have been calling the base module. The base module may have edges pointing to external terms, but those terms are not present in the main ontology. This means you would load to-base and po-base and they would slot together and you would have no overlay of differing versions.

Not all ontologies provide the base module yet but you can ping me to ping others to move this along.

**Comment by cmungall** [comment]
@kimrutherford has experience/pain working with go-plus, which brings in portions of other ontologies

**Comment by bradfordcondon** [comment]
>The base module may have edges pointing to external terms, but those terms are not present in the main ontology
Maybe I'm misunderstanding you, but what about cases where intermediate node terms are used in the ontology?

**Comment by kimrutherford** [comment]
The PomBase Chado DB uses option 2 but the ontologies we use only borrow terms in straightforward ways.  We haven't had any problems with this plan.

> @kimrutherford has experience/pain working with go-plus, which brings in portions of other ontologies

We use go-basic for Chado loading.  We use go-plus for Canto because it includes the taxon restrictions.  We use owltools to produce a pombe-only OBO file for loading into Canto.  At the moment we only load relations between terms in the same namespace and we only load the relations Canto needs.  So we have it easy.  We have no problems with borrowed terms in Canto either.
